templates/common/infra/terraform/applicationinsights.tf

```diff
@@ -0,0 +1,1241 @@
+terraform {
+  required_providers {
+    azurerm = {
+      version = "~>3.18.0"
+      source  = "hashicorp/azurerm"
+    }
+    azurecaf = {
+      source  = "aztfmod/azurecaf"
+      version = "1.2.15"
+    }
+  }
+}
+# ------------------------------------------------------------------------------------------------------
+# DEPLOY APPLICATION INSIGHTS
+# ------------------------------------------------------------------------------------------------------
+resource "azurecaf_name" "applicationinsights_name" {
+  name          = var.resource_token
+  resource_type = "azurerm_application_insights"
+  random_length = 0
+  clean_input   = true
+}
+
+resource "azurerm_application_insights" "applicationinsights" {
+  name                = azurecaf_name.applicationinsights_name.result
+  location            = var.location
+  resource_group_name = var.rg_name
+  application_type    = "web"
+  workspace_id        = var.workspace_id
+  tags                = var.tags
+}
+
+data "azurerm_subscription" "current" {}
+
+resource "azurerm_portal_dashboard" "board" {
+  name                 = "dash-${var.resource_token}"

```

Comment:
I think you said all prefixes come from TF somewhere, right?

Reply:
this resource type is not supported yet. [Here](https://github.com/aztfmod/terraform-provider-azurecaf) is the supported resources list.

Reply:
Ok